confluent_kafka/src/Consumer.c

```diff
@@ -508,6 +589,22 @@ static PyMethodDef Consumer_methods[] = {
 	  "  :raises: KafkaException\n"
 	  "\n"
 	},
+	{ "get_watermark_offsets", (PyCFunction)Consumer_get_watermark_offsets,
```

Comment:
Assignment and watermark offsets seem quite different re: the PR. The assignment seems pretty uncontroversial -- we know what we want exposed here.

Is the watermark the same? If we're consistent everywhere, that's fine, just asking because I think this is something we want to be consistent about across languages and I'm not sure we've fully specified the semantics of. Even the cached versions seem like they may need clarification (e.g., what is the cache duration?). The high/low offsets seem clearer, but I think in the Java client there may be some desire to make any request for offsets a bit more on-demand compared to the high offset provided with each fetch request.

/cc @hachikuji since I think we may one day (hopefully in the near future!) want to provide this for the Java clients as `KafkaBasedLog` and similar classes could greatly benefit from a real LEO API...